### ISSUE TITLE
Child component resolver does not wait for its parent component resolver to finish

### ISSUE THREAD

**Issue by LyubomirVetskovProgress** [angular/angular#12148]
**I'm submitting a ...**
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
I have two nested components - parent and child, both with separate resolvers. Here is the routing config:

```javascript
const routes = [
{
    path: '',
    redirectTo: '/parent/1',
    pathMatch: 'full'
},
{
    path: 'parent',
    resolve: {
        parent: ParentResolver
    },
    children: [
	    { path: '',  component: ParentComponent },
	    {
	    	path: ':child',
	    	component: ChildComponent,
	    	resolve: {
	            child: ChildResolver
	        }
	    }
    ]
}];

```

When entering directly the URL corresponding to the child component ` '/parent/1',` , the two resolvers start resolving concurrently. This leads to a conflict, when `ChildResolver` depends on a state modification, made by its parent component's `ParentResolver`.

**Expected behavior**
When entering directly the URL corresponding to the child component ` '/parent/1',` , the expected behavior is that the `ParentResolver` will get resolved first, and after that `ChildResolver` resolution will take place. 

**Minimal reproduction of the problem with instructions**

1. Go to [plunker](http://plnkr.co/edit/qt2SmbWjxE8tQMejK3jm?p=preview)
2. Open browser's console and you will see that the `ChildResolver` has finished before `ParentResolver`. If you look at their implementation you will see, that the `ChildResolver` resolution timeout is shorter than the resolution timeout of `ParentResolver`, which means that the resolvers are concurrent.

**What is the motivation / use case for changing the behavior?**

This looks as a valid routing configuration that should be supported.

**Please tell us about your environment:**

* **Angular version:** 2.0.X

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by pkozlowski-opensource** [comment]
Isn't it the same as #12032 that got fixed in master via 1681e4f57f4c4fa45a556502df0533131c2d64b4?

**Comment by LyubomirVetskovProgress** [comment]
It seems like this fix indeed has fixed the issue. Thanks for the reference!
